plugins/aiops/pkg/gateway/metricai.go

```diff
@@ -0,0 +1,365 @@
+		if errors.Is(err, nats.ErrNoKeysFound) {
+			return &metricai.MetricAIIdList{}, nil
+		}
+		return nil, status.Errorf(codes.NotFound, "Failed to ListJobs for metricAI: %v", err)
+	}
+	return &metricai.MetricAIIdList{Items: jobs}, nil
+}
+
+// list keys in the natsKV JobRun bucket
+func (p *AIOpsPlugin) ListJobRuns(ctx context.Context, jobId *metricai.MetricAIId) (*metricai.MetricAIIdList, error) {
+	ctxca, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	metricAIKeyValue, err := p.metricAIRunKv.GetContext(ctxca)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to ListJobRuns for metricAI: %v", err)
+	}
+	jobruns, err := metricAIKeyValue.Keys()
+	if err != nil {
+		if errors.Is(err, nats.ErrNoKeysFound) {
+			return &metricai.MetricAIIdList{}, nil
+		}
+		return nil, status.Errorf(codes.NotFound, "Failed to ListJobRuns for metricAI: %v", err)
+	}
+	var jobRunIdArray []string
+	// use jobId.Id + jobRunDelimiter to uniquely identify jobrun IDs for different jobs.
+	for _, j := range jobruns {
+		if strings.HasPrefix(j, jobId.Id+jobRunDelimiter) {
+			jobRunIdArray = append(jobRunIdArray, j)
+		}
+
+	}
+	return &metricai.MetricAIIdList{Items: jobRunIdArray}, nil
+}
+
+func (p *AIOpsPlugin) RunJob(ctx context.Context, jobRequest *metricai.MetricAIId) (*metricai.MetricAIRunJobResponse, error) {
+	// run a job. Post a request to the python metric-ai service.
+	ctxca, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	// make the http request with context
+	url := serverUrl + "/run_job/" + jobRequest.Id
+	req, err := http.NewRequestWithContext(ctxca, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to form httprequest to RunJob for metricAI: %v", err)
+	}
+	resp, err := p.httpClient.Do(req)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to make httprequest to RunJob for metricAI: %v", err)
+	}
+	defer resp.Body.Close()
+	var res map[string]interface{}
+
+	if err := json.NewDecoder(resp.Body).Decode(&res); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to unmarshal response of SubmitJobRequest for metricAI: %v", err)
+	}
+	return &metricai.MetricAIRunJobResponse{JobRunId: res["JobRunId"].(string), SubmittedTime: time.Now().String(), Status: res["Status"].(string)}, nil
+}
+
+func (p *AIOpsPlugin) CreateJob(ctx context.Context, jobRequest *metricai.MetricAICreateJobRequest) (*metricai.MetricAIAPIResponse, error) {
+	// use the info provided by user to create a job
+	// Info includes: job's name, the cluseter_id, a list of namespaces to watch.
+	ctxca, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	metricAIKeyValue, err := p.metricAIJobKv.GetContext(ctxca)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to ListJobRuns for metricAI: %v", err)
+	}
+	jid := strings.ToLower(jobRequest.JobId)
+
+	if jid == "" { // id can't be empty
+		return nil, &RequestError{
```

Comment:
Using this struct definition won't convert codes in the GRPC response, for that we should prefer using grpc `codes` package like in the other apis in this file. 

stylistic nit : for input validation we typically use "github.com/rancher/opni/pkg/validation" and implement the Validator interface, see for example https://github.com/rancher/opni/blob/main/pkg/apis/control/v1/validate.go

Reply:
good point. Added code using the validator interface.